File: src/main/java/nz/co/searchwellington/commentfeeds/CommentFeedDetectorService.java
package nz.co.searchwellington.commentfeeds;

import nz.co.searchwellington.commentfeeds.detectors.CommentFeedDetector;
import org.apache.log4j.Logger;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Component;

@Component
public class CommentFeedDetectorService {

    private static Logger log = Logger.getLogger(CommentFeedDetectorService.class);
    
	private CommentFeedDetector[] detectors;
        
	@Autowired
	public CommentFeedDetectorService(CommentFeedDetector... detectors) {
		this.detectors = detectors;
		log.info("Autowired " + detectors.length + " comment detectors: " + detectors);
	}
	
	public boolean isCommentFeedUrl(String url) {
        for (CommentFeedDetector detector : detectors) {
        	log.debug(detector.getClass().getName() + " is checking is comment feed url: " + url);
            if (detector.isValid(url)) {
            	log.debug(detector.getClass().getName() + " detected comment feed url: " + url);
                return true;
            }
        }    
        return false;        
    }
	
}

File: src/main/scala/nz/co/searchwellington/commentfeeds/CommentFeedDetectorService.scala
package nz.co.searchwellington.commentfeeds

import nz.co.searchwellington.commentfeeds.detectors.CommentFeedDetector
import nz.co.searchwellington.model.Resource
import org.apache.commons.logging.LogFactory
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.stereotype.Component

import java.net.URL

@Component class CommentFeedDetectorService @Autowired()(detectors: Array[CommentFeedDetector]) {

  private val log = LogFactory.getLog(classOf[CommentFeedDetectorService])

  log.info("Autowired " + detectors.length + " comment detectors: " + detectors.toSeq.map(_.getClass.getSimpleName).mkString(", "))

  def isCommentFeedUrl(url: URL, source: Resource): Boolean = {
    detectors.find(_.isValid(url, source)).exists { d =>
      log.info(d.getClass.getName + " detected comment feed url: " + url)
      true
    }
  }

}